nic.lua

```diff
@@ -48,13 +48,33 @@ minetest.register_node("digistuff:nic", {
 			action = function(pos,node,channel,msg)
 					local meta = minetest.get_meta(pos)
 					if meta:get_string("channel") ~= channel then return end
-					if type(msg) ~= "string" then return end
+					local url
+					local headers
+					local parse_json = false
+					-- parse message
+					if type(msg) == "string" then
+						-- simple string data
+						url = msg
+					elseif type(msg) == "table" and type(msg.url) == "string" then
+						-- config object
+						url = msg.url
+						headers = msg.headers
```

Comment:
Should check type of headers before using it.

Reply:
Also contents, if you send `headers = {false}` it will crash server.
Probably something like:
```lua
if type(msg.headers) == "table" and #msg.headers > 0 and #msg.headers < 50 then
  headers = {}
  for _, header in ipairs(msg.headers) do
    header = (tostring(header)):gmatch("%C+")()
    if header ~= "" then
      table.insert(headers, header)
    end
end